python/hidet/option.py

```diff
@@ -177,11 +194,40 @@ def register_hidet_options():
     )
     register_option(
         name='cuda.arch',
-        type_hint='Optional[str]',
-        default_value=None,
-        description='The CUDA architecture to compile the kernels for (e.g., "sm_70"). None for auto-detect.',
+        type_hint='str',
+        default_value='auto',
+        description='The CUDA architecture to compile the kernels for (e.g., "sm_70"). "auto" for auto-detect.',
     )
 
```

Comment:
Consider putting the logic into seperate functions:
```
config_file_path = ...
if os.path.exists(config_file_path):
    _load_config(config_file_path)
else:
    _write_default_config(config_file_path)
```
where in `_write_default_config` function, we create a default toml doc from the registered options.